File: scratch/whitelabel.py
import os
import re

workspace = r"c:\Users\Aidan\Downloads\signalform-briefs-main (2)"

# Define replacements
replacements = [
    (re.compile(r"Signalform Studio", re.IGNORECASE), "Briefs Studio"),
    (re.compile(r"Signalform", re.IGNORECASE), "Briefs Studio"),
    (re.compile(r"Signal Form", re.IGNORECASE), "Briefs Studio")
]

exclude_dirs = {"node_modules", ".git", ".agent", "brain", ".system_generated"}
include_exts = {".html", ".json", ".js", ".toml", ".md", ".rtf", ".css"}

modified_count = 0
file_count = 0

print("Starting whitelabel search and replace...")

for root, dirs, files in os.walk(workspace):
    # Skip excluded directories in place
    dirs[:] = [d for d in dirs if d not in exclude_dirs]
    
    for file in files:
        ext = os.path.splitext(file)[1].lower()
        if ext not in include_exts:
            continue
            
        filepath = os.path.join(root, file)
        
        try:
            with open(filepath, 'r', encoding='utf-8', errors='ignore') as f:
                content = f.read()
                
            new_content = content
            made_changes = False
            
            for pattern, replacement in replacements:
                if pattern.search(new_content):
                    new_content = pattern.sub(replacement, new_content)
                    made_changes = True
            
            if made_changes:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print(f"Whitelabeled: {os.path.relpath(filepath, workspace)}")
                modified_count += 1
            
            file_count += 1
        except Exception as e:
            print(f"Error processing {file}: {e}")

print(f"\nWhitelabel complete! Processed {file_count} files, updated {modified_count} files.")

File: scratch/whitelabel.js
import fs from 'fs/promises';
import path from 'path';

const workspace = "c:\\Users\\Aidan\\Downloads\\Briefs Studio-briefs-main (2)";

const replacements = [
  { pattern: /Briefs Studio/gi, replacement: "Briefs Studio" },
  { pattern: /Briefs Studio/gi, replacement: "Briefs Studio" },
  { pattern: /Briefs Studio/gi, replacement: "Briefs Studio" }
];

const excludeDirs = new Set(["node_modules", ".git", ".agent", "brain", ".system_generated"]);
const includeExts = new Set([".html", ".json", ".js", ".toml", ".md", ".rtf", ".css"]);

let modifiedCount = 0;
let fileCount = 0;

async function walk(dir) {
  const files = await fs.readdir(dir, { withFileTypes: true });
  
  for (const file of files) {
    const resPath = path.join(dir, file.name);
    
    if (file.isDirectory()) {
      if (excludeDirs.has(file.name)) continue;
      await walk(resPath);
    } else {
      const ext = path.extname(file.name).toLowerCase();
      if (!includeExts.has(ext)) continue;
      
      try {
        let content = await fs.readFile(resPath, 'utf-8');
        let newContent = content;
        let madeChanges = false;
        
        for (const { pattern, replacement } of replacements) {
          if (pattern.test(newContent)) {
            newContent = newContent.replace(pattern, replacement);
            madeChanges = true;
          }
        }
        
        if (madeChanges) {
          await fs.writeFile(resPath, newContent, 'utf-8');
          console.log(`Whitelabeled: ${path.relative(workspace, resPath)}`);
          modifiedCount++;
        }
        fileCount++;
      } catch (err) {
        console.error(`Error reading ${resPath}:`, err.message);
      }
    }
  }
}

async function run() {
  console.log("Starting Node.js whitelabel search and replace...");
  await walk(workspace);
  console.log(`\nWhitelabel complete! Processed ${fileCount} files, updated ${modifiedCount} files.`);
}

run();
